contrib/log_correlation/stackdriver/src/main/java/io/opencensus/contrib/logcorrelation/stackdriver/OpenCensusTraceLoggingEnhancer.java

```diff
@@ -66,19 +74,31 @@
   /**
    * Constructor to be called by Stackdriver logging.
    *
-   * <p>This constructor looks up the {@link SpanSelection} from the environment using the property
-   * name {@value SPAN_SELECTION_PROPERTY_NAME}. It looks for a {@link java.util.logging} property
-   * or a system property, giving preference to the logging property.
+   * <p>This constructor looks up the project ID and {@link SpanSelection SpanSelection} from the
+   * environment. It uses the project ID returned from {@code
+   * com.google.cloud.ServiceOptions.getDefaultProjectId()}, unless the ID is overridden by the
+   * property {@value PROJECT_ID_PROPERTY_NAME}. It looks up the {@code SpanSelection} using the
+   * property {@value SPAN_SELECTION_PROPERTY_NAME}. Each property can be specified with a {@link
+   * java.util.logging} property or a system property, with preference given to the logging
+   * property.
    */
   public OpenCensusTraceLoggingEnhancer() {
-    this(lookUpSpanSelectionProperty());
+    this(lookUpProjectId(), lookUpSpanSelectionProperty());
```

Comment:
Optional: instead of just projectID, cache: `"projects/" + projectId + "/traces/"`

Reply:
Thanks!  I forgot I could also cache the whole prefix.